src/Package/Impl/Sql/DragDrop/DataObject.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Windows;
+using Microsoft.Common.Core;
+using Microsoft.Languages.Editor.DragDrop;
+
+namespace Microsoft.VisualStudio.R.Package.Sql.DragDrop {
+    public static class DataObject {
+        public static string GetPlainText(this IDataObject dataObject) {
+            var flags = dataObject.GetFlags();
+            if ((flags & DataObjectFlags.ProjectItems) != 0) {
+                var item = dataObject.GetProjectItems().FirstOrDefault();
+                if (item != null && Path.GetExtension(item.FileName).EqualsIgnoreCase(".r")) {
+                    return GetFileContent(item.FileName).Replace("'", "''");
+                }
+            }
+            return string.Empty;
+        }
+
+        private static string GetFileContent(string file) {
+            try {
+                return File.ReadAllText(file).Trim();
```

Comment:
VS stores everything as UTF-8 without BOM by default. So it depends where the file was created. In this case it most probably is created by RTVS. However, yes, it might come from elsewhere theoretically.